Added suppport for Label Attributes

shubhamnarlawar77 opened the pull request:
@marxin : review request || @andikleen : review request

shubhamnarlawar77 committed:
Added suppport for Label Attributes

marxin commented on src/Statement.cpp:
```diff
@@ -302,6 +316,11 @@ Statement::make_random(CGContext &cg_context,
 		break;
 	}
 
+	if(!label_attr_generate){
```
I'm wondering whether we can find a better place to initialize the `label_attr_generator`. What about in constructor of `StatementFilter`? It can be a singleton instance if I'm correct.

shubhamnarlawar77 replied:
> I'm wondering whether we can find a better place to initialize the `label_attr_generator`. What about in constructor of `StatementFilter`? It can be a singleton instance if I'm correct.

The constructor of `StatementFilter` is being called more than once. 

Should I move below code to `StatementFilter` constructor?
`if(!label_attr_generate){
                InitializeLabelAttributes();
                label_attr_generate = true;
        }
`

marxin replied:
Yes please, make it a static variable and initialize it just once.

shubhamnarlawar77 replied:
Does it look fine now?

marxin reviewed (changes requested):
In general it's fine. Just one comment..

marxin commented on src/Statement.cpp:
```diff
@@ -84,11 +84,25 @@
 const Statement* Statement::failed_stm;
 
+static bool label_attr_generate;
+AttributeGenerator label_attr_generator;
```
Please do not use global variables, but use static class variables.

shubhamnarlawar77 replied:
Done. It is now static variable of class `StatementFilter`.

marxin replied:
Sorry, I can't see `label_attr_generator` being a static variable of the class.

shubhamnarlawar77 replied:
> Sorry, I can't see `label_attr_generator` being a static variable of the class.

My apologies. I misunderstood previous comment about doing static variable. 

Now, I have declared `label_attr_generator` as static. Does your above comment means `static object of the class?` Please correct me if I am wrong?

marxin replied:
> Now, I have declared `label_attr_generator` as static. Does your above comment means `static object of the class?` Please correct me if I am wrong?

 It means put both variables into the `class` and make them `static`.

shubhamnarlawar77 commented:
One of them label_attr_generator is object of class AttributeGenerator. Do you want to put it class itself?

marxin commented:
> One of them label_attr_generator is object of class AttributeGenerator. Do you want to put it class itself?

You use it in `Statement`, so I would expect to be static of this class.

shubhamnarlawar77 commented:
> > One of them label_attr_generator is object of class AttributeGenerator. Do you want to put it class itself?
> 
> You use it in `Statement`, so I would expect to be static of this class.

Done.
1. Moved `label_attr_generator` to Statement class
2. Moved `label_attr_generate` to StatementFilter class

Is it fine now?

marxin commented:
> Is it fine now?

Yes, go on and merge it.
